### ISSUE TITLE
RandomForest classifier became slower between 1.10 and 1.11

### ISSUE THREAD

**Issue by hmeine** [ukoethe/vigra#353]
I updated vigra in order to see whether the RandomForest got any faster since the 1.10.0 release.  Unfortunately, it became at least 20% slower in my benchmarks.  I thought I should bring it up here for discussion.

**Comment by ukoethe** [comment]
Do you refer to training or prediction? In any case, we didn't change much in the code, so the problem might be easy to see in the diff.

**Comment by hmeine** [comment]
Prediction.  I loaded the same classifier from disk.  (Apparently, the HDF5 layout did not change – that's nice.)

**Comment by ukoethe** [comment]
I did a quick scan of the diff and don't see any code changes in the relevant functions between 1.10 and 1.11 (except for a few explicit casts you added to silence warnings). Please be more specific.

**Comment by hmeine** [comment]
I did the same, and also only found changes related to compiler warnings. (And unrelated things like stopping criteria.)
What I did is stopping the time it took to classify a certain volume before and after updating vigra, and that was roughly 98…100 sec. with 1.10.0 and 120 sec. after the update.  Next step will be to reproduce these measurements, preferably without MeVisLab around it, but e.g. using vigranumpy.  I will report back later ("later" meaning not today).
